src/main/java/VASSAL/tools/swing/SwingUtils.java

```diff
@@ -53,115 +72,300 @@ public static AffineTransform descaleTransform(AffineTransform t) {
     /*
      * @return whether the event effectively has Control down
      */
-    boolean isControlDown(MouseEvent e);
+    boolean isControlDown(MouseEvent e);    
+
+    /*
+     * @return whether this is key/mouse combo that brings up context menus on our platform (whether or not it brings them up right this moment)
+     */
+    boolean isContextMouseButtonDown(MouseEvent e);
+        
+    /*
+     * @return whether this the key/mouse combo for clicking on things to select them. (normally just plain old Left Mouse Button - but on Mac only when not pretending to be right button)
+     */
+    boolean isVanillaLeftButtonDown(MouseEvent e);
+    
+    /*
+     * @return whether this is key/mouse combo that toggles items in and out of selections on our platform (e.g. Ctrl+LeftClick on non-Mac)
+     */
+    boolean isSelectionToggle(MouseEvent e);
+    
+    /*
+     * @returns translation of keystroke from local system to Vassal (to handle Mac platform support)
+     */
+    KeyStroke getKeySystemToVassal (KeyStroke k);
+    
+    /*
+     * @returns translation of keystroke from Vassal to local system (to handle Mac platform support)
+     */
+    KeyStroke getKeyVassalToSystem (KeyStroke k);    
   }
 
+  
+  // Pass through -- presently handles all systems except Macs
```

Comment:
You can turn this into javadoc, then the IDE's will show neat popup menus with this text when hovering over the class name.